### PR TITLE
[PORT] Adds the Syndie-Lad to the game.

### PR DESCRIPTION
**Pull request by Moccha-Bee**
## About The Pull Request
Traitors can now set their uplink of choice to the Syndie-Lad, a portable arm mounted computer that tracks objectives and stores your uplink. Equipping it and pressing C will open the radial menu, or just hold it in your hand and press Z (assuming you're on hotkey mode). It also explodes, if you really need it to.

Fallout memes.

From:https://github.com/HippieStation/HippieStation/pull/11825

## Why It's Good For The Game
A item that instantly makes you valid, and serves to be both an uplink and tracker for objectives. And can be set as the default uplink in game preferneces.

## Changelog
:cl:
add: Added: Syndielad
/:cl:

### CODE REVIEW

**Comment by Moccha-Bee** [comment]
Cringe, compiled without errors....

**Comment by Moccha-Bee** [comment]
travis machine broke

**Comment by MarkSuckerberg** [comment]
bro moccha machine broke ~~and doesn't know how to check travis logs~~
```
code/game/objects/items/syndielad.dm, line 2, column 9:

warning: macro redefined: SYNDIELAD

- 1:9: previous definition of SYNDIELAD

code/game/objects/items/syndielad.dm, line 3, column 9:

warning: macro redefined: SYNDIELAD

- 2:9: previous definition of SYNDIELAD

code/game/objects/items/syndielad.dm, line 4, column 9:

warning: macro redefined: SYNDIELAD

- 3:9: previous definition of SYNDIELAD
```

File: code/game/objects/items/syndielad.dm [hunk 1]
```diff
@@ -0,0 +1,300 @@
+#define SYNDIELAD-POWEROFF "poweroff"
+#define SYNDIELAD-UPLINK "uplink"
+#define SYNDIELAD-PINPOINTER "pinpointer"
+#define SYNDIELAD-DETONATE "detonate"
```
Comment (MarkSuckerberg) [inline code/game/objects/items/syndielad.dm]: it doesn't like these

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/game/objects/items/syndielad.dm (hunk 1 above)
Comment (MarkSuckerberg) [reply code/game/objects/items/syndielad.dm]: remove the `SYNDIELAD-` before them

File: code/game/objects/items/syndielad.dm [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+			. += "Extreme danger. Arming signal detected. Time remaining: [bomb.get_time_left()]."
+
+/obj/item/clothing/gloves/syndielad/nuke/toggle_pinpointer(mob/living/user)
+	syndilad_pinactive = !syndilad_pinactive
+	if(syndilad_pinactive)
+		START_PROCESSING(SSfastprocess, src)
+	else
+		syndilad_target = null
+		STOP_PROCESSING(SSfastprocess, src)
+	update_icon()
```
Comment (MarkSuckerberg) [inline code/game/objects/items/syndielad.dm]: and add `#undef POWEROFF, #undef UPLINK`, etc, etc... down here

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/game/objects/items/syndielad.dm (hunk 2 above)
Comment (Moccha-Bee) [reply code/game/objects/items/syndielad.dm]: Made all the changes

File: code/game/objects/items/syndielad.dm (hunk 1 above)
Comment (ike709) [reply code/game/objects/items/syndielad.dm]: Better solution is to replace `-` with `_`. Without the prefix, these are all generic define names that could cause conflicts at some point.

File: code/game/objects/items/syndielad.dm (hunk 2 above)
Comment (ike709) [reply code/game/objects/items/syndielad.dm]: Eh, you did undef them though so it's probably not a big deal.

**Review by ike709 — CHANGES_REQUESTED** [review]
Only gave this a quick glance, but at the very minimum all sleeps need to be converted to addtimers.

**Comment by ike709** [comment]
That isn't how addtimer works. At all.

**Comment by Toxici11i** [comment]
pretty sure this was ported w/o all relevant fixes. strongly against adding this, as the syndielad provides shock proof, and heatproof gloves roundstart for absolutely 0 cost, and acts as a minibomb if set off

**Pull request closed without merging**